Week6/BatterySDK.tsx

```diff
@@ -0,0 +1,143 @@
+import { useEffect, useState, useCallback } from "react";
+import * as Battery from "expo-battery";
+import { StyleSheet, Text, View } from "react-native";
+import { Accelerometer } from "expo-sensors";
+import { Accel } from "./Accelerometer";
+
+export default function BatterySDK() {
+  const [batteryLevel, setBatteryLevel] = useState<number | null>(null);
+  const [subscription, setSubscription] = useState<Battery.Subscription | null>(
+    null
+  );
+  const batLevel = batteryLevel !== null ? Math.round(100 * batteryLevel) : 0;
+  const { x, y, z } = Accel();
+  const shake = Math.sqrt(x * x + y * y + z * z);
+  const batPercent = `${batLevel}%`;
+
+  const _subscribe = async () => {
+    const batteryLevel = await Battery.getBatteryLevelAsync();
+    setBatteryLevel(batteryLevel);
+
+    setSubscription(
+      Battery.addBatteryLevelListener(({ batteryLevel }) => {
+        setBatteryLevel(batteryLevel);
+        console.log("batteryLevel changed!", batteryLevel);
+      })
+    );
+  };
+
+  const _unsubscribe = useCallback(() => {
+    subscription && subscription.remove();
+    setSubscription(null);
+  }, [subscription]);
+
+  // useEffect(() => {
+  //   _subscribe();
+  //   return () => _unsubscribe();
+  // }, [_unsubscribe]);
+
+  useEffect(() => {
+    if (shake > 1.1) {
+      setBatteryLevel((batteryLevel) =>
+        Math.min((batteryLevel as any) + 0.01, +1)
+      );
+    }
+  }, [shake]);
+
+  const batColor = (batLevel: number) => {
+    let batteryColor = "#b00d23";
+
+    if (batLevel >= 50) {
```

Comment:
You could simplify this a little. Here I initialize to green since that's the widest range and most likely to be correct and then we only change the color.
```
const batColor = (batLevel: number) => {
  let batteryColor = '#5BC236';

  if (batLevel < 20) {
    batteryColor = '#b00d23';
  } else if (batLevel < 50) {
    batteryColor = '#FFF224';
  }

  return batteryColor;
};
```